sqlglot/expressions.py

```diff
@@ -233,6 +234,25 @@ def transform(self, fun, *args, copy=True, **kwargs):
         return new_node
 
 
+class Term(Expression):
+    @classmethod
+    def ensure_expression(cls, value):
+        if isinstance(value, Expression):
+            return value
+        if value is None:
+            return Null()
+        if isinstance(value, date):
+            return Literal(this=value.isoformat(), is_string=True)
+        if isinstance(value, str):
+            return Literal(this=value, is_string=True)
+        if isinstance(value, (int, float)):
+            return Literal(this=value, is_string=False)
+        raise ValueError(f"Unable to convert value to Expression: {value}")
+
+    def eq(self, other):
```

Comment:
is it possible to override

def __equals__ ? so that we can do x == y

Reply:
We already use `__eq__` in `Expression`.

But again, we won't need this anymore.